lib/ex_aws/s3/utils.ex

```diff
@@ -60,6 +60,23 @@ defmodule ExAws.S3.Utils do
     |> format_and_take(param_list)
   end
 
+  def apply_client_options([], client), do: client
+  def apply_client_options(opts, %{config: config} = client) do
```

Comment:
This is unnecessary. You can already override the client by using the `ExAws.S3.Impl` functions which is the correct way to do this.